--- drupy/ported/includes/path.inc	2008-04-03 01:31:11.000000000 -0700
+++ drupal/includes/path.inc	2008-04-07 00:25:11.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: path.inc,v 1.20 2008/02/18 16:49:23 dries Exp $
+// $Id: path.inc,v 1.21 2008/04/06 18:50:11 dries Exp $
 
 /**
  * @file
@@ -135,7 +135,7 @@ function drupal_get_normal_path($path, $
 /**
  * Return a component of the current Drupal path.
  *
- * When viewing a page at the path "admin/content/types", for example, arg(0)
+ * When viewing a page at the path "admin/build/types", for example, arg(0)
  * would return "admin", arg(1) would return "content", and arg(2) would return
  * "types".
  *
