--- drupy/ported/includes/path.inc	2008-06-27 16:42:26.000000000 -0700
+++ drupal/includes/path.inc	2008-06-28 11:24:30.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: path.inc,v 1.22 2008/04/14 17:48:33 dries Exp $
+// $Id: path.inc,v 1.24 2008/06/24 22:12:15 dries Exp $
 
 /**
  * @file
@@ -46,15 +46,20 @@ function drupal_init_path() {
 function drupal_lookup_path($action, $path = '', $path_language = '') {
   global $language;
   // $map is an array with language keys, holding arrays of Drupal paths to alias relations
-  static $map = array(), $no_src = array();
+  static $map = array(), $no_src = array(), $count;
 
   $path_language = $path_language ? $path_language : $language->language;
 
+  // Use $count to avoid looking up paths in subsequent calls if there simply are no aliases
+  if (!isset($count)) {
+    $count = db_result(db_query('SELECT COUNT(pid) FROM {url_alias}'));
+  }
+
   if ($action == 'wipe') {
     $map = array();
     $no_src = array();
   }
-  elseif (module_exists('path') && $path != '') {
+  elseif ($count > 0 && $path != '') {
     if ($action == 'alias') {
       if (isset($map[$path_language][$path])) {
         return $map[$path_language][$path];
@@ -233,5 +238,5 @@ function drupal_match_path($path, $patte
   if (!isset($regexps[$patterns])) {
     $regexps[$patterns] = '/^(' . preg_replace(array('/(\r\n?|\n)/', '/\\\\\*/', '/(^|\|)\\\\<front\\\\>($|\|)/'), array('|', '.*', '\1' . preg_quote(variable_get('site_frontpage', 'node'), '/') . '\2'), preg_quote($patterns, '/')) . ')$/';
   }
-  return preg_match($regexps[$patterns], $path);
+  return (bool)preg_match($regexps[$patterns], $path);
 }
