--- drupy/ported/modules/user/user.module	2008-07-19 03:01:21.000000000 -0700
+++ drupal/modules/user/user.module	2008-08-13 22:50:16.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: user.module,v 1.911 2008/06/27 07:25:11 dries Exp $
+// $Id: user.module,v 1.914 2008/08/08 19:48:43 dries Exp $
 
 /**
  * @file
@@ -653,7 +653,7 @@ function user_user($type, &$edit, &$acco
     $account->content['summary']['member_for'] =  array(
       '#type' => 'user_profile_item',
       '#title' => t('Member for'),
-      '#value' => format_interval(time() - $account->created),
+      '#markup' => format_interval(time() - $account->created),
     );
   }
   if ($type == 'form' && $category == 'account') {
@@ -701,7 +701,7 @@ function user_login_block() {
     $items[] = l(t('Create new account'), 'user/register', array('attributes' => array('title' => t('Create a new user account.'))));
   }
   $items[] = l(t('Request new password'), 'user/password', array('attributes' => array('title' => t('Request new password via e-mail.'))));
-  $form['links'] = array('#value' => theme('item_list', $items));
+  $form['links'] = array('#markup' => theme('item_list', $items));
   return $form;
 }
 
@@ -1367,8 +1367,11 @@ function user_login_submit($form, &$form
 function user_external_login_register($name, $module) {
   global $user;
 
-  $user = user_load(array('name' => $name));
-  if (!isset($user->uid)) {
+  $existing_user = user_load(array('name' => $name));
+  if (isset($existing_user->uid)) {
+    $user = $existing_user;
+  }
+  else {
     // Register this new user.
     $userinfo = array(
       'name' => $name,
@@ -1493,7 +1496,7 @@ function user_edit_form(&$form_state, $u
     $form['picture'] = array('#type' => 'fieldset', '#title' => t('Picture'), '#weight' => 1);
     $picture = theme('user_picture', (object)$edit);
     if ($edit['picture']) {
-      $form['picture']['current_picture'] = array('#value' => $picture);
+      $form['picture']['current_picture'] = array('#markup' => $picture);
       $form['picture']['picture_delete'] = array('#type' => 'checkbox', '#title' => t('Delete picture'), '#description' => t('Check this box to delete your current picture.'));
     }
     else {
@@ -1552,12 +1555,12 @@ function user_delete($edit, $uid) {
   $account = user_load(array('uid' => $uid));
   sess_destroy_uid($uid);
   _user_mail_notify('status_deleted', $account);
+  module_invoke_all('user', 'delete', $edit, $account);
   db_query('DELETE FROM {users} WHERE uid = %d', $uid);
   db_query('DELETE FROM {users_roles} WHERE uid = %d', $uid);
   db_query('DELETE FROM {authmap} WHERE uid = %d', $uid);
   $variables = array('%name' => $account->name, '%email' => '<' . $account->mail . '>');
   watchdog('user', 'Deleted user: %name %email.', $variables, WATCHDOG_NOTICE);
-  module_invoke_all('user', 'delete', $edit, $account);
 }
 
 /**
@@ -2335,7 +2338,7 @@ function user_register() {
 
   // Display the registration form.
   if (!$admin) {
-    $form['user_registration_help'] = array('#value' => filter_xss_admin(variable_get('user_registration_help', '')));
+    $form['user_registration_help'] = array('#markup' => filter_xss_admin(variable_get('user_registration_help', '')));
   }
 
   // Merge in the default user edit fields.
