--- drupy/ported/includes/file.inc	2008-07-19 03:01:21.000000000 -0700
+++ drupal/includes/file.inc	2008-08-13 22:50:14.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: file.inc,v 1.126 2008/06/18 03:36:23 dries Exp $
+// $Id: file.inc,v 1.127 2008/07/05 18:34:29 dries Exp $
 
 /**
  * @file
@@ -139,12 +139,12 @@ function file_check_directory(&$director
   // Check if directory exists.
   if (!is_dir($directory)) {
     if (($mode & FILE_CREATE_DIRECTORY) && @mkdir($directory)) {
-      drupal_set_message(t('The directory %directory has been created.', array('%directory' => $directory)));
       @chmod($directory, 0775); // Necessary for non-webserver users.
     }
     else {
       if ($form_item) {
         form_set_error($form_item, t('The directory %directory does not exist.', array('%directory' => $directory)));
+        watchdog('file system', 'The directory %directory does not exist.', array('%directory' => $directory), WATCHDOG_ERROR);
       }
       return FALSE;
     }
@@ -152,10 +152,7 @@ function file_check_directory(&$director
 
   // Check to see if the directory is writable.
   if (!is_writable($directory)) {
-    if (($mode & FILE_MODIFY_PERMISSIONS) && @chmod($directory, 0775)) {
-      drupal_set_message(t('The permissions of directory %directory have been changed to make it writable.', array('%directory' => $directory)));
-    }
-    else {
+    if (($mode & FILE_MODIFY_PERMISSIONS) && !@chmod($directory, 0775)) {
       form_set_error($form_item, t('The directory %directory is not writable', array('%directory' => $directory)));
       watchdog('file system', 'The directory %directory is not writable, because it does not have the correct permissions set.', array('%directory' => $directory), WATCHDOG_ERROR);
       return FALSE;
