--- /cvs/drupal/drupal/includes/menu.inc	2008/03/25 14:03:02	1.255.2.10
+++ /cvs/drupal/drupal/includes/menu.inc	2008/04/09 21:11:44	1.255.2.11
@@ -1,5 +1,5 @@
 <?php
-// $Id: menu.inc,v 1.255.2.9 2008/02/27 12:12:01 goba Exp $
+// $Id: menu.inc,v 1.255.2.10 2008/03/25 14:03:02 goba Exp $
 
 /**
  * @file
@@ -2275,9 +2275,10 @@
         if (!isset($item['tab_root']) && !$parent['_tab']) {
           $item['tab_root'] = $parent_path;
         }
-        // If a callback is not found, we try to find the first parent that
-        // has a callback.
-        if (!isset($item['access callback']) && isset($parent['access callback'])) {
+        // If an access callback is not found for a default local task we use
+        // the callback from the parent, since we expect them to be identical.
+        // In all other cases, the access parameters must be specified.
+        if (($item['type'] == MENU_DEFAULT_LOCAL_TASK) && !isset($item['access callback']) && isset($parent['access callback'])) {
           $item['access callback'] = $parent['access callback'];
           if (!isset($item['access arguments']) && isset($parent['access arguments'])) {
             $item['access arguments'] = $parent['access arguments'];
