--- drupy/ported/includes/menu.inc	2008-06-27 16:42:26.000000000 -0700
+++ drupal/includes/menu.inc	2008-06-28 11:24:30.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: menu.inc,v 1.275 2008/06/12 20:49:39 dries Exp $
+// $Id: menu.inc,v 1.278 2008/06/25 09:12:24 dries Exp $
 
 /**
  * @file
@@ -622,7 +622,7 @@ function _menu_translate(&$router_item, 
   $router_item['href'] = implode('/', $link_map);
   $router_item['options'] = array();
   _menu_check_access($router_item, $map);
-  
+
   // For performance, don't localize an item the user can't access.
   if ($router_item['access']) {
     _menu_item_localize($router_item, $map);
@@ -1251,28 +1251,28 @@ function menu_get_names($reset = FALSE) 
  * Return an array containing the names of system-defined (default) menus.
  */
 function menu_list_system_menus() {
-  return array('navigation', 'primary-links', 'secondary-links');
+  return array('navigation', 'main-menu', 'secondary-menu');
 }
 
 /**
- * Return an array of links to be rendered as the Primary links.
+ * Return an array of links to be rendered as the Main menu.
  */
-function menu_primary_links() {
-  return menu_navigation_links(variable_get('menu_primary_links_source', 'primary-links'));
+function menu_main_menu() {
+  return menu_navigation_links(variable_get('menu_main_menu_source', 'main-menu'));
 }
 
 /**
  * Return an array of links to be rendered as the Secondary links.
  */
-function menu_secondary_links() {
+function menu_secondary_menu() {
 
   // If the secondary menu source is set as the primary menu, we display the
   // second level of the primary menu.
-  if (variable_get('menu_secondary_links_source', 'secondary-links') == variable_get('menu_primary_links_source', 'primary-links')) {
-    return menu_navigation_links(variable_get('menu_primary_links_source', 'primary-links'), 1);
+  if (variable_get('menu_secondary_menu_source', 'secondary-menu') == variable_get('menu_main_menu_source', 'main-menu')) {
+    return menu_navigation_links(variable_get('menu_main_menu_source', 'main-menu'), 1);
   }
   else {
-    return menu_navigation_links(variable_get('menu_secondary_links_source', 'secondary-links'), 0);
+    return menu_navigation_links(variable_get('menu_secondary_menu_source', 'secondary-menu'), 0);
   }
 }
 
@@ -1574,10 +1574,10 @@ function menu_set_active_trail($new_trai
         $curr = FALSE;
       }
       else {
-        // Move to the child link if it's in the active trail.
-        if ($curr['below'] && $curr['link']['in_active_trail']) {
+        // Add the link if it's in the active trail, then move to the link below.
+        if ($curr['link']['in_active_trail']) {
           $trail[] = $curr['link'];
-          $tree = $curr['below'];
+          $tree = $curr['below'] ? $curr['below'] : array();
         }
         list($key, $curr) = each($tree);
       }
