--- drupy/ported/includes/theme.inc	2008-06-27 16:42:26.000000000 -0700
+++ drupal/includes/theme.inc	2008-06-28 11:24:30.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: theme.inc,v 1.426 2008/06/06 01:50:20 dries Exp $
+// $Id: theme.inc,v 1.428 2008/06/25 09:12:24 dries Exp $
 
 /**
  * @file
@@ -864,8 +864,8 @@ function theme_get_settings($key = NULL)
     'logo_path'                     =>  '',
     'default_favicon'               =>  1,
     'favicon_path'                  =>  '',
-    'primary_links'                 =>  1,
-    'secondary_links'               =>  1,
+    'main_menu'                     =>  1,
+    'secondary_menu'                =>  1,
     'toggle_logo'                   =>  1,
     'toggle_favicon'                =>  1,
     'toggle_name'                   =>  1,
@@ -874,8 +874,8 @@ function theme_get_settings($key = NULL)
     'toggle_mission'                =>  1,
     'toggle_node_user_picture'      =>  0,
     'toggle_comment_user_picture'   =>  0,
-    'toggle_primary_links'          =>  1,
-    'toggle_secondary_links'        =>  1,
+    'toggle_main_menu'              =>  1,
+    'toggle_secondary_menu'         =>  1,
   );
 
   if (module_exists('node')) {
@@ -1838,8 +1838,8 @@ function template_preprocess_page(&$vari
   $variables['logo']              = theme_get_setting('logo');
   $variables['messages']          = $variables['show_messages'] ? theme('status_messages') : '';
   $variables['mission']           = isset($mission) ? $mission : '';
-  $variables['primary_links']     = theme_get_setting('toggle_primary_links') ? menu_primary_links() : array();
-  $variables['secondary_links']   = theme_get_setting('toggle_secondary_links') ? menu_secondary_links() : array();
+  $variables['main_menu']     = theme_get_setting('toggle_main_menu') ? menu_main_menu() : array();
+  $variables['secondary_menu']   = theme_get_setting('toggle_secondary_menu') ? menu_secondary_menu() : array();
   $variables['search_box']        = (theme_get_setting('toggle_search') ? drupal_get_form('search_theme_form') : '');
   $variables['site_name']         = (theme_get_setting('toggle_name') ? variable_get('site_name', 'Drupal') : '');
   $variables['site_slogan']       = (theme_get_setting('toggle_slogan') ? variable_get('site_slogan', '') : '');
@@ -1965,7 +1965,8 @@ function template_preprocess_node(&$vari
     $variables['picture'] = '';
   }
   // Clean up name so there are no underscores.
-  $variables['template_files'][] = 'node-' . $node->type;
+  $variables['template_files'][] = 'node-' . str_replace('_', '-', $node->type);
+  $variables['template_files'][] = 'node-' . $node->nid;
 }
 
 /**
