--- drupy/ported/includes/theme.maintenance.inc	2008-06-27 16:42:26.000000000 -0700
+++ drupal/includes/theme.maintenance.inc	2008-06-28 11:24:30.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: theme.maintenance.inc,v 1.13 2008/04/28 09:25:26 dries Exp $
+// $Id: theme.maintenance.inc,v 1.14 2008/06/25 09:12:24 dries Exp $
 
 /**
  * @file
@@ -252,8 +252,8 @@ function template_preprocess_maintenance
   $variables['logo']              = theme_get_setting('logo');
   $variables['messages']          = $variables['show_messages'] ? theme('status_messages') : '';
   $variables['mission']           = '';
-  $variables['primary_links']     = array();
-  $variables['secondary_links']   = array();
+  $variables['main_menu']         = array();
+  $variables['secondary_menu']    = array();
   $variables['search_box']        = '';
   $variables['site_name']         = (theme_get_setting('toggle_name') ? variable_get('site_name', 'Drupal') : '');
   $variables['site_slogan']       = (theme_get_setting('toggle_slogan') ? variable_get('site_slogan', '') : '');
