--- drupy/ported/includes/path.inc	2008-04-07 00:42:36.000000000 -0700
+++ drupal/includes/path.inc	2008-04-14 12:37:37.000000000 -0700
@@ -1,5 +1,5 @@
 <?php
-// $Id: path.inc,v 1.21 2008/04/06 18:50:11 dries Exp $
+// $Id: path.inc,v 1.22 2008/04/14 17:48:33 dries Exp $
 
 /**
  * @file
@@ -231,7 +231,7 @@ function drupal_match_path($path, $patte
   static $regexps;
 
   if (!isset($regexps[$patterns])) {
-    $regexps[$patterns] = '/^('. preg_replace(array('/(\r\n?|\n)/', '/\\\\\*/', '/(^|\|)\\\\<front\\\\>($|\|)/'), array('|', '.*', '\1'. preg_quote(variable_get('site_frontpage', 'node'), '/') .'\2'), preg_quote($patterns, '/')) .')$/';
+    $regexps[$patterns] = '/^(' . preg_replace(array('/(\r\n?|\n)/', '/\\\\\*/', '/(^|\|)\\\\<front\\\\>($|\|)/'), array('|', '.*', '\1' . preg_quote(variable_get('site_frontpage', 'node'), '/') . '\2'), preg_quote($patterns, '/')) . ')$/';
   }
   return preg_match($regexps[$patterns], $path);
 }
